In [1]:
# import dependencies
import pandas as pd

In [2]:
# file being used
file = "Resources/purchase_data.csv"

In [3]:
# Store File into Pandas DataFrame
purchase_data = pd.read_csv(file)

In [4]:
# Number of Players
total_players = purchase_data["SN"].nunique()
total_players

576

In [5]:
print("------ PLAYER COUNT --------")
print("----------------------------")
print("Total Number of Players: 576")

------ PLAYER COUNT --------
----------------------------
Total Number of Players: 576


In [6]:
# PURCHASING ANALYSIS (Total)
    # Number of Unique Items
unique_items = purchase_data["Item Name"].nunique()
    # Average Purchase Price
avg_price = purchase_data["Price"].mean()
    # Total Number of Purchases
total_purchases = purchase_data.count()
    # Total Revenue
total_revenue = purchase_data["Price"].sum()

In [7]:
print("----- PURCHASING ANALYSIS ----")
print("------------------------------")
print("Number of Unique Items: 179")
print("Average Purchase Price: $3.05")
print("Total Number of Purchases: 780")
print("Total Revenue: $2,379.77")

----- PURCHASING ANALYSIS ----
------------------------------
Number of Unique Items: 179
Average Purchase Price: $3.05
Total Number of Purchases: 780
Total Revenue: $2,379.77


In [8]:
# Gender Demographics
count_gender = purchase_data.groupby("Gender")["SN"].nunique()
percent_gender = (count_gender/576) * 100
gender_demo = pd.DataFrame({"Gender by Count": count_gender,
                           "Gender by Percent": percent_gender})
gender_demo

Gender by Count  Gender by Percent
Gender                                                   
Female                              81          14.062500
Male                               484          84.027778
Other / Non-Disclosed               11           1.909722

In [9]:
# PURCHASING ANALYSIS (GENDER)
    #Purchase Count
purchase_count = purchase_data.groupby("Gender")["Item Name"].count()
purchase_count

Gender
Female                   113
Male                     652
Other / Non-Disclosed     15
Name: Item Name, dtype: int64

In [10]:
 # Average Purchase Price
avg_purchase_price = purchase_data.groupby("Gender")["Price"].mean()
avg_purchase_price

Gender
Female                   3.203009
Male                     3.017853
Other / Non-Disclosed    3.346000
Name: Price, dtype: float64

In [11]:
# Total Purchase Value
total_purchase_value = purchase_data.groupby("Gender")["Price"].sum()
total_purchase_value

Gender
Female                    361.94
Male                     1967.64
Other / Non-Disclosed      50.19
Name: Price, dtype: float64

In [12]:
# Age Demographics
age = purchase_data["Age"].value_counts()
age

20    99
22    70
24    67
23    67
21    62
25    59
15    35
30    35
16    30
18    26
19    23
17    22
35    14
33    14
26    14
29    13
27    10
10     9
7      9
34     9
38     9
32     8
8      8
11     7
31     7
37     7
39     6
40     6
9      6
12     6
36     5
28     5
13     4
44     2
41     2
14     2
43     1
42     1
45     1
Name: Age, dtype: int64

In [13]:
# Put in Bins
bins = [20,25,30,35,40,45]
labels = ["Twenty to Twenty-Four","Twenty-Five to Twenty-Nine","Thirty to Thirty-Four","Thirty-Five to Thirty-Nine","Fourty to Fourty-Four",]

In [14]:
purchase_data["Age Ranges"] = pd.cut(purchase_data["Age"],bins, labels=labels)

In [15]:
#Purchase Count
purchase_count = purchase_data.groupby("Age Ranges")
purchase_count["Purchase ID"].count()

Age Ranges
Twenty to Twenty-Four         325
Twenty-Five to Twenty-Nine     77
Thirty to Thirty-Four          52
Thirty-Five to Thirty-Nine     33
Fourty to Fourty-Four           7
Name: Purchase ID, dtype: int64

In [16]:
#Average Purchase Price
purchase_count["Price"].mean()

Age Ranges
Twenty to Twenty-Four         3.020431
Twenty-Five to Twenty-Nine    2.875584
Thirty to Thirty-Four         2.994423
Thirty-Five to Thirty-Nine    3.404545
Fourty to Fourty-Four         3.075714
Name: Price, dtype: float64

In [17]:
#Average Purchase Total per Person by Age Group
purchase_count["Price"].mean()

Age Ranges
Twenty to Twenty-Four         3.020431
Twenty-Five to Twenty-Nine    2.875584
Thirty to Thirty-Four         2.994423
Thirty-Five to Thirty-Nine    3.404545
Fourty to Fourty-Four         3.075714
Name: Price, dtype: float64

Three Observable Trends Based On Data:
1. 20 to 24 year olds are, by 248, the largest age range for the game.
2. There are 403 more male than female players, and 392 more male players than female and other / undisclosed players.
3. There are only 484 players, and an income of $2,379.77 from 179 purchases, which is only 37% of the player base buying items. Some of those buyers buy frequently which means less than 37% of the player base buys items. Considering the cost to run game servers the company should consider marketing promotions to players or expanding the player base to increase revenue - unless the game is run by one person on a home server internet connection, in which case he's doing pretty okay!